components/hierarchical-view/hierarchical-view-mixin.js

```diff
@@ -0,0 +1,380 @@
+import { findComposedAncestor, isComposedAncestor } from '../../helpers/dom.js';
+import { getNextFocusable, getPreviousFocusable } from '../../helpers/focus.js';
+import ResizeObserver from 'resize-observer-polyfill/dist/ResizeObserver.es.js';
+
+export const HierarchicalViewMixin = superclass => class extends superclass {
+
+	static get properties() {
+		return {
+			shown: { type: Boolean, reflect: true },
+			childView: { type: Boolean, reflect: true, attribute: 'child-view' },
+			isAttached: { type: Boolean, reflect: true, attribute: 'is-attached' },
+			isHierarchicalView: { type: Boolean, reflect: true, attribute: 'is-hierarchical-view' }
```

Comment:
It could also probably be private eh, since it's only needed by this mixin?

Reply:
Yeah, I think the `isHierarchicalView` is only called from within this code and we don't expect any consumer code to call this. 

I think `childView` is similar in this regard, although it is also reflected since there are some styles that target `[child-view]`. 

... for private reflected properties, I've started to prefix them so that it's clear they are not part of the public API and not to be touched by consumers. (ex. dialog-mixin has a `_state` attribute which is targeted by some CSS). Usually I try to contain that stuff in the shadow-DOM, but from time to time it's necessary to surface an attribute on the host. Thoughts on that practice?

Reply:
Seems fine yeah. Agreed that if you can put those classes in the shadow DOM that's preferred... the element internals stuff that's coming will help solve this too.

Reply:
So `isHierarchicalView` is used in the `menu-item-mixin` so I don't think we would want it private in this case. It does look like it does not need to be reflected though, so I'll remove that and fix the name.